backend/api/Services/InspectionAreaService.cs

```diff
@@ -127,6 +134,83 @@ public async Task<IEnumerable<InspectionArea>> ReadByInstallation(
                 .FirstOrDefaultAsync();
         }
 
+        public bool MissionTasksAreInsideInspectionAreaPolygon(
+            List<MissionTask> missionTasks,
+            InspectionArea inspectionArea
+        )
+        {
+            if (string.IsNullOrEmpty(inspectionArea.AreaPolygonJson))
+            {
+                return true; // No polygon defined, so all tasks are inside
+            }
+
+            var inspectionAreaPolygon = JsonSerializer.Deserialize<InspectionAreaPolygon>(
+                inspectionArea.AreaPolygonJson
+            );
+
+            if (inspectionAreaPolygon == null)
+            {
+                return true; // Invalid polygon, so all tasks are inside
+            }
+
+            var inspectionAreaPositions = inspectionAreaPolygon.Positions;
+            var inspectionAreaZMin = inspectionAreaPolygon.ZMin;
+            var inspectionAreaZMax = inspectionAreaPolygon.ZMax;
+            foreach (var missionTask in missionTasks)
+            {
+                var robotPosition = missionTask.RobotPose.Position;
+                if (
+                    !IsPositionInsidePolygon(
+                        inspectionAreaPositions,
+                        robotPosition,
+                        inspectionAreaZMin,
+                        inspectionAreaZMax
+                    )
+                )
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static bool IsPositionInsidePolygon(
+            List<XYPosition> polygon,
+            Position position,
+            double zMin,
+            double zMax
+        )
+        {
+            var x = position.X;
+            var y = position.Y;
+            var z = position.Z;
+
+            if (z < zMin || z > zMax)
+            {
+                return false;
+            }
+
+            // Ray-casting algorithm for checking if the point is inside the polygon
+            var inside = false;
+            for (int i = 0, j = polygon.Count - 1; i < polygon.Count; j = i++)
+            {
+                var xi = polygon[i].X;
+                var yi = polygon[i].Y;
+                var xj = polygon[j].X;
+                var yj = polygon[j].Y;
+
+                var intersect =
+                    ((yi > y) != (yj > y)) && (x < (xj - xi) * (y - yi) / (yj - yi) + xi);
+                if (intersect)
+                {
+                    inside = !inside;
+                }
+            }
```

Comment:
Should we not count how many intersections and check if n_intersections % 2 == 1?
https://medium.com/@girishajmera/exploring-algorithms-to-determine-points-inside-or-outside-a-polygon-038952946f87

Reply:
Or wait, I get it. It is just flipped every time there is an intersection. This works

Reply:
Yeah, just implemented the ray casting algorithm